### PR TITLE
Sb 04

### PR DESCRIPTION
**Pull request by Alex-Korzun**

### CODE REVIEW

File: src/main/java/com/banking/bankservice/controller/AccountController.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    @GetMapping("/{accountNumber}")
+    public Double getBalanceByAccountNumber(@PathVariable String accountNumber) {
+        return accountService.getAccountByAccountNumber(accountNumber).getBalance();
+    }
+
+    @GetMapping("/history/{accountNumber}")
+    public List<TransactionResponseDto> getHistoryByAccountNumber(
+            @PathVariable String accountNumber, @RequestParam(defaultValue = "1") Integer page,
```
Comment (KseniiaMakarova) [inline src/main/java/com/banking/bankservice/controller/AccountController.java]: pages start with 0
let's also set `required = false`

File: src/main/java/com/banking/bankservice/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    @Autowired
+    public UserController(UserService userService, UserMapper userMapper) {
+        this.userService = userService;
+        this.userMapper = userMapper;
+    }
+
+    @PostMapping
+    public void create(@RequestBody UserRequestDto userRequestDto) {
+        User user = userMapper.fromDto(userRequestDto);
+        userService.save(user);
```
Comment (KseniiaMakarova) [inline src/main/java/com/banking/bankservice/controller/UserController.java:41]: let's set USER role to this user

File: src/main/java/com/banking/bankservice/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @PutMapping("/{userId}")
+    public UserResponseDto update(@RequestBody UserRequestDto userRequestDto,
+                                  @PathVariable Long userId) {
+        User user = userMapper.fromDto(userRequestDto);
+        user.setId(userId);
```
Comment (KseniiaMakarova) [inline src/main/java/com/banking/bankservice/controller/UserController.java:49]: set roles that were previously held by this user as well

File: src/main/java/com/banking/bankservice/model/dto/request/AccountRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package com.banking.bankservice.model.dto.request;
+
+import com.banking.bankservice.model.Currency;
+import lombok.Data;
+
+@Data
+public class AccountRequestDto {
+    private String accountNumber;
+    private Long userID;
```
Comment (KseniiaMakarova) [inline src/main/java/com/banking/bankservice/model/dto/request/AccountRequestDto.java]:
```suggestion
    private Long userId;
```

File: src/main/java/com/banking/bankservice/mapper/impl/AccountMapperImpl.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import org.springframework.stereotype.Component;
+
+@Component
+public class AccountMapperImpl implements AccountMapper {
+    @Override
+    public Account fromDto(AccountRequestDto accountRequestDto) {
+        Account account = new Account();
+        account.setAccountNumber(accountRequestDto.getAccountNumber());
+        account.setBalance(accountRequestDto.getBalance());
+        account.setActive(true);
```
Comment (KseniiaMakarova) [inline src/main/java/com/banking/bankservice/mapper/impl/AccountMapperImpl.java]: do we set it twice?

File: src/main/java/com/banking/bankservice/mapper/impl/AccountMapperImpl.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        account.setCurrencyType(Currency.valueOf(accountRequestDto.getCurrencyType().toString()));
```
Comment (KseniiaMakarova) [inline src/main/java/com/banking/bankservice/mapper/impl/AccountMapperImpl.java]: can we simplify this?

File: src/main/java/com/banking/bankservice/util/DataInitializer.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    @PostConstruct
+    public void init() {
+        User admin = new User();
+        admin.setName("admin");
+        admin.setPhoneNumber("0661237698");
+        admin.setPassword("admin");
+        admin.setDateOfBirth(LocalDate.of(1989, 4, 25));
+        Role adminRole = new Role();
+        adminRole.setRoleName(Role.Roles.ADMIN);
+        roleService.save(adminRole);
```
Comment (KseniiaMakarova) [inline src/main/java/com/banking/bankservice/util/DataInitializer.java:32]: let's also save USER role

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/banking/bankservice/mapper/impl/AccountMapperImpl.java (hunk 6 above)
Comment (Alex-Korzun) [reply src/main/java/com/banking/bankservice/mapper/impl/AccountMapperImpl.java]: to be honest, I don't know how

File: src/main/java/com/banking/bankservice/mapper/impl/AccountMapperImpl.java [hunk 8]
```diff
@@ -14,7 +14,6 @@ public Account fromDto(AccountRequestDto accountRequestDto) {
         Account account = new Account();
         account.setAccountNumber(accountRequestDto.getAccountNumber());
         account.setBalance(accountRequestDto.getBalance());
-        account.setActive(true);
         account.setCurrencyType(Currency.valueOf(accountRequestDto.getCurrencyType().toString()));
```
Comment (KseniiaMakarova) [inline src/main/java/com/banking/bankservice/mapper/impl/AccountMapperImpl.java]: account.setCurrencyType(accountRequestDto.getCurrencyType());

File: src/main/java/com/banking/bankservice/mapper/impl/AccountMapperImpl.java [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+import com.banking.bankservice.model.dto.request.AccountRequestDto;
+import com.banking.bankservice.model.dto.response.AccountResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AccountMapperImpl implements AccountMapper {
+    @Override
+    public Account fromDto(AccountRequestDto accountRequestDto) {
+        Account account = new Account();
+        account.setAccountNumber(accountRequestDto.getAccountNumber());
```
Comment (KseniiaMakarova) [inline src/main/java/com/banking/bankservice/mapper/impl/AccountMapperImpl.java:23]: let's set User to account?

File: src/main/java/com/banking/bankservice/mapper/impl/AccountMapperImpl.java [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+        return account;
+    }
+
+    @Override
+    public AccountResponseDto toDto(Account account) {
+        return AccountResponseDto.builder()
+                .accountNumber(account.getAccountNumber())
+                .balance(account.getBalance())
+                .currency(account.getCurrencyType().toString())
+                .build();
```
Comment (KseniiaMakarova) [inline src/main/java/com/banking/bankservice/mapper/impl/AccountMapperImpl.java:38]: not all fields are set

File: src/main/java/com/banking/bankservice/util/DataInitializer.java [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+        userService.save(admin);
+
+        Role userRole = new Role();
+        roleService.save(userRole);
```
Comment (KseniiaMakarova) [inline src/main/java/com/banking/bankservice/util/DataInitializer.java:37]: is everything ok here?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/banking/bankservice/mapper/impl/AccountMapperImpl.java (hunk 8 above)
Comment (Alex-Korzun) [reply src/main/java/com/banking/bankservice/mapper/impl/AccountMapperImpl.java]: Okay, sorry. It was stupid question

File: src/main/java/com/banking/bankservice/mapper/impl/AccountMapperImpl.java (hunk 9 above)
Comment (Alex-Korzun) [reply src/main/java/com/banking/bankservice/mapper/impl/AccountMapperImpl.java:23]: done

File: src/main/java/com/banking/bankservice/mapper/impl/AccountMapperImpl.java (hunk 10 above)
Comment (Alex-Korzun) [reply src/main/java/com/banking/bankservice/mapper/impl/AccountMapperImpl.java:38]: Did I manage boolean `isActive` correctly?

File: src/main/java/com/banking/bankservice/util/DataInitializer.java (hunk 11 above)
Comment (Alex-Korzun) [reply src/main/java/com/banking/bankservice/util/DataInitializer.java:37]: added `RoleName`

File: src/main/java/com/banking/bankservice/mapper/impl/AccountMapperImpl.java (hunk 10 above)
Comment (KseniiaMakarova) [reply src/main/java/com/banking/bankservice/mapper/impl/AccountMapperImpl.java:38]: yep)

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**